### PR TITLE
Gray out label when disabled

### LINKED ISSUE #7634: On disable switch make the text grayed out
## Test plan
See https://github.com/brave/browser-laptop/pull/12455

## Original issue description
- Did you search for similar issues before submitting this one?

- Describe the issue you encountered:

- Platform (Win7, 8, 10? macOS? Linux distro?):

- Brave Version (revision SHA):

- Steps to reproduce:
    1.
    2.
    3.
    
- Actual result:

- Expected result:

- Will the steps above reproduce in a fresh profile? If not what other info can be added?

- Is this an issue in the currently released version?

- Can this issue be consistently reproduced?

- Extra QA steps:
    1.
    2.
    3.

- Screenshot if needed:

- Any related issues:

### PR DESCRIPTION
**Pull request by user512**
Fix #7634

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
go to about:preferences#plugins without installing Pepper Flash and observe label is gray out.
screen shot 2018-01-01 at 2 27 36 pm

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/renderer/components/common/switchControl.js [hunk 1]
```diff
@@ -131,7 +131,8 @@ class SwitchControl extends ImmutableComponent {
               styles.switchControl__text__label,
               styles.switchControl__text__label_right,
               this.props.small && styles.switchControl__text__label_small,
-              this.props.customStyleTextRight
+              this.props.customStyleTextRight,
+              this.props.disabled && styles.switchControl__text_disabled
```
Comment (luixxiul) [inline app/renderer/components/common/switchControl.js]: @user512 would you mind moving `this.props.customStyleTextRight` under this line as it is expected to override the inherited properties?

File: app/renderer/components/common/switchControl.js [hunk 2]
```diff
@@ -131,6 +131,7 @@ class SwitchControl extends ImmutableComponent {
               styles.switchControl__text__label,
               styles.switchControl__text__label_right,
               this.props.small && styles.switchControl__text__label_small,
+              this.props.disabled && styles.switchControl__text_disabled,
```
Comment (user512) [inline app/renderer/components/common/switchControl.js]: @luixxiul, thank you for the feedback.
I am not very familiar with styling in this app or styling in React, would you mind telling me how does order affect styling? I tried reading https://github.com/brave/browser-laptop/blob/master/docs/style.md but can't find how order affect styling.

Would be great if you can suggested some reading about styling in React, thanks 😄

Comment (cezaraugusto) [reply app/renderer/components/common/switchControl.js]: hey @user512. The order affects styles the same way as CSS cascading works. The last rule overrides the previous so in your changes, the disabled interface would override other styles, which is the correct and expected behavior. Nice work!

Comment (cezaraugusto) [reply app/renderer/components/common/switchControl.js]: maybe this could be the last conditional but I think it's ok as-is. Thanks!

**Comment by srirambv** [comment]
@bsclifton @user512 Is this fix only for Flash switch or all switch in general when off?

**Comment by bsclifton** [comment]
@srirambv this should be for all places where it's used (in a disabled state). The test plan captures the only known occurrence... but if you set the `disabled` field on any future ones, it would work like this too.

**Comment by srirambv** [comment]
@bsclifton none of the switches which are turned off shows gray out. The flash text does it when there is no flash installed on the machine. Otherwise all other text just looks normal.

**Comment by bsclifton** [comment]
@srirambv per the test plan, it should only be grayed out when Flash is not installed on the machine (disabled). Because you can't turn something on if it's not installed 😛 So you just verified this is working as expected 😄 👍

**Comment by srirambv** [comment]
That works but the original issue was intended to show grayed out text when the switch was turned off. So i was expecting to see all text in gray which were turned off. Basically like this https://github.com/brave/browser-laptop/issues/7634#issuecomment-285217620

**Pull request merged**